V0.44.0 changelog: Pauli-based computation (#8842)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

  [(#8748)](https://github.com/PennyLaneAI/pennylane/pull/8748)

<h4>Pauli-based computation </h4>
  
  New tools dedicated to fault-tolerant quantum computing (FTQC) research based on the Pauli-based
  computation framework are now available! The functionalities are designed to allow rapid and flexible
  design, optimization and inspectability of quantum circuits at the logical level.
  
  Pauli product rotations (PPRs) and Pauli product measurements (PPMs) are the building blocks for
  quantum computation under the Pauli-based model. In FTQC based on surface code architectures, to
  save on classical compilation times, it is useful to write common gates and subroutines directly

>   in this representation.

Author: Oh, nice way to condense the information 👍

* Writing circuits in terms of `Pauli product measurements <https://pennylane.ai/compilation/pauli-product-measurement>`_
  (PPMs) in PennyLane is now possible with the new :func:`~.pauli_measure` function.
  Using this function in tandem with :class:`~.PauliRot` to represent PPRs unlocks surface-code
  fault-tolerant quantum computing research spurred from `A Game of Surface Codes <http://arxiv.org/abs/1808.02892>`_.
  [(#8461)](https://github.com/PennyLaneAI/pennylane/pull/8461)
  [(#8631)](https://github.com/PennyLaneAI/pennylane/pull/8631)
  [(#8623)](https://github.com/PennyLaneAI/pennylane/pull/8623)
  [(#8663)](https://github.com/PennyLaneAI/pennylane/pull/8663)
  [(#8692)](https://github.com/PennyLaneAI/pennylane/pull/8692)

  The new :func:`~.pauli_measure` function is currently only for analysis on the ``null.qubit`` device,
  which allows for resource tracking with :func:`~.specs` and circuit inspection with :func:`~.drawer.draw`.

  In the following example, a measurement of the ``XY`` Pauli product on wires ``0`` and ``2`` is performed

>   using :func:`~.pauli_measure`, followed by application of a :class:`~.PauliX` gate conditional on

Reviewer: I can't comment on every line, but maybe just this:

> Using :func:`~.pauli_measure` in a circuit is similar to :func:`~.measure` (a mid-circuit measurement), but requires that a ``pauli_word`` be specified for the measurement basis:

    expval(PauliZ): 1
  ```

* Additionally, Catalyst compilation passes designed for Pauli-based computation are now available in PennyLane, 

>   providing accessibility for logical compilation research by directly integrating with 

Reviewer: ```suggestion
* Several qjit-compatible compilation passes designed for Pauli-based computation are now available with this release, and are designed to work directly with
```

  :func:`~.pauli_measure` and :class:`~.PauliRot` operations.
  [(#8609)](https://github.com/PennyLaneAI/pennylane/pull/8609)
  [(#8764)](https://github.com/PennyLaneAI/pennylane/pull/8764)
  [(#8762)](https://github.com/PennyLaneAI/pennylane/pull/8762)
  
  These compilation passes include:

  * The new :func:`~pennylane.transforms.gridsynth` compilation pass to perform rapid Clifford+T
  decomposition with :func:`pennylane.qjit`. This pass discretizes ``RZ`` and ``PhaseShift``

>   gates to either the Clifford+T basis or to the Pauli product rotation (PPR) basis.

Reviewer: ```suggestion
  * :func:`~.transforms.gridsynth`: This pass decomposes :math:`Z`-basis rotations and and :class:`~.PhaseShift`
  gates to either the Clifford+T basis or to other PPRs.
```

    The :func:`~pennylane.transforms.gridsynth` function accepts an ``epsilon`` arguments that determines
    the maximum permissible operator norm error per rotation gate:

> 

Reviewer: ```suggestion
```

    ```python
    @qml.qnode(qml.device("lightning.qubit", wires=1))
    def circuit(x):
        qml.Hadamard(0)
        qml.RZ(x, 0)
        qml.PhaseShift(x * 0.2, 0)
        return qml.state()

    gridsynth_circuit = qml.transforms.gridsynth(circuit, epsilon=1e-4)

>     qjitted_circuit = qml.qjit(gridsynth_circuit)

Reviewer: ```suggestion
```

    ```
    ```pycon
    >>> qjitted_circuit(1.1)
    [0.6028324 -0.3695921j  0.50763281+0.49224355j]
    ```

  * Five transforms for decomposing Clifford+T gates into PPRs and PPMs and further optimizing a circuit down to PPMs,
  namely, func:`pennylane.transforms.to_ppr`, :func:`pennylane.transforms.commute_ppr`, :func:`pennylane.transforms.merge_ppr_ppm`,
  :func:`pennylane.transforms.ppr_to_ppm`, and :func:`pennylane.transforms.ppm_compilation`. Additionally, the

>   :func:`pennylane.transforms.reduce_t_depth` transform to reduce the depth of T gates. 

Reviewer: ```suggestion
  * Five transforms for compiling Clifford+T gates, PPRs, and/or PPMs, including func:`~.transforms.to_ppr`, :func:`~.transforms.commute_ppr`, :func:`~.transforms.merge_ppr_ppm`,
  :func:`~.transforms.ppr_to_ppm`, and :func:`~.transforms.ppm_compilation`,
  :func:`pennylane.transforms.reduce_t_depth`, and :func:`~.transforms.decompose_arbitrary_ppr`.
```

  
    ```python
    from pennylane.transforms import to_ppr

    qml.capture.enable()

> 

Reviewer: ```suggestion
```

    @qml.qjit(target="mlir")
    @to_ppr
    @qml.qnode(qml.device("null.qubit", wires=2))
    def circuit():
        qml.H(0)
        qml.CNOT([0, 1])
        qml.T(0)
        return qml.expval(qml.Z(0))
    ```
    ```pycon
    >>> print(qml.specs(circuit, level="all", compute_depth=False)()['resources'])
    ```

* It is not always necessary to go via the Clifford+T to PPR representations. For the following operators

>   shortcut decompositions directly to the :class:`~.PauliRot` gate have been supported:

Reviewer: ```suggestion
* Directly decomposing Clifford+T gates and other small gates into PPRs is possible via :func:`~.transforms.decompose` and enabling graph-based decompositions with :func:`~.decomposition.enable_graph`. This allows for bypassing the :func:`~.transforms.clifford_t_decomposition`, which can be expensive in some cases.
```
